From 3e663ee5cc7f0babbda32a8ddebb90fd4b613b1b Mon Sep 17 00:00:00 2001
From: Guilhem IMBERTON <guilhem.imberton@intel.com>
Date: Thu, 3 Oct 2013 13:20:26 +0200
Subject: [PORT FROM MAIN][BUILD] Fix required modules dependency in prebuilds

BZ: 142768

Fix dependencies for BUILD_HOST_PREBUILT and BUILD_MULTI_PREBUILT

    Category: Build
    Domain: AOSP
    Origin: Internal
    Upstream-Candidate: Yes

Change-Id: I76d927f6c3becaa1258646d08f9dc452c418f811
Category: device enablement
Domain: AOSP
Origin: internal
Upstream-Candidate: no
Signed-off-by: Guilhem IMBERTON <guilhem.imberton@intel.com>
Reviewed-on: http://android.intel.com:8080/136075
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 core/multi_prebuilt.mk |   30 ++++++++++++++++++++----------
 1 files changed, 20 insertions(+), 10 deletions(-)

diff --git a/core/multi_prebuilt.mk b/core/multi_prebuilt.mk
index bc85cea..dfcf2db 100644
--- a/core/multi_prebuilt.mk
+++ b/core/multi_prebuilt.mk
@@ -28,6 +28,7 @@ prebuilt_static_java_libraries := $(LOCAL_PREBUILT_STATIC_JAVA_LIBRARIES)
 prebuilt_is_host := $(LOCAL_IS_HOST_MODULE)
 prebuilt_module_tags := $(LOCAL_MODULE_TAGS)
 prebuilt_strip_module := $(LOCAL_STRIP_MODULE)
+prebuilt_required_modules := $(LOCAL_REQUIRED_MODULES)
 
 
 ifndef multi_prebuilt_once
@@ -37,10 +38,12 @@ multi_prebuilt_once := true
 # $(2): IS_HOST_MODULE
 # $(3): MODULE_CLASS
 # $(4): MODULE_TAGS
-# $(5): OVERRIDE_BUILT_MODULE_PATH
-# $(6): UNINSTALLABLE_MODULE
-# $(7): BUILT_MODULE_STEM
-# $(8): LOCAL_STRIP_MODULE
+# $(5): LOCAL_REQUIRED_MODULES
+# $(6): OVERRIDE_BUILT_MODULE_PATH
+# $(7): UNINSTALLABLE_MODULE
+# $(8): BUILT_MODULE_STEM
+# $(9): LOCAL_STRIP_MODULE
+
 #
 # Elements in the file list may be bare filenames,
 # or of the form "<modulename>:<filename>".
@@ -55,8 +58,9 @@ $(foreach t,$(1), \
   $(eval LOCAL_IS_HOST_MODULE := $(2)) \
   $(eval LOCAL_MODULE_CLASS := $(3)) \
   $(eval LOCAL_MODULE_TAGS := $(4)) \
-  $(eval OVERRIDE_BUILT_MODULE_PATH := $(5)) \
-  $(eval LOCAL_UNINSTALLABLE_MODULE := $(6)) \
+  $(eval LOCAL_REQUIRED_MODULES := $(5)) \
+  $(eval OVERRIDE_BUILT_MODULE_PATH := $(6)) \
+  $(eval LOCAL_UNINSTALLABLE_MODULE := $(7)) \
   $(eval tw := $(subst :, ,$(strip $(t)))) \
   $(if $(word 3,$(tw)),$(error $(LOCAL_PATH): Bad prebuilt filename '$(t)')) \
   $(if $(word 2,$(tw)), \
@@ -66,8 +70,8 @@ $(foreach t,$(1), \
     $(eval LOCAL_MODULE := $(basename $(notdir $(t)))) \
     $(eval LOCAL_SRC_FILES := $(t)) \
    ) \
-  $(if $(7), \
-    $(eval LOCAL_BUILT_MODULE_STEM := $(7)) \
+  $(if $(8), \
+    $(eval LOCAL_BUILT_MODULE_STEM := $(8)) \
    , \
     $(if $(word 2,$(tw)), \
       $(eval LOCAL_BUILT_MODULE_STEM := $(LOCAL_MODULE)$(suffix $(LOCAL_SRC_FILES))) \
@@ -77,7 +81,7 @@ $(foreach t,$(1), \
    ) \
   $(eval LOCAL_MODULE_SUFFIX := $(suffix $(LOCAL_SRC_FILES))) \
   $(if $(filter user,$(TARGET_BUILD_VARIANT)), \
-    $(eval LOCAL_STRIP_MODULE := $(8))) \
+    $(eval LOCAL_STRIP_MODULE := $(9))) \
   $(eval include $(BUILD_PREBUILT)) \
  )
 endef
@@ -90,6 +94,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     STATIC_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     true)
 
@@ -98,6 +103,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     SHARED_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     $($(if $(prebuilt_is_host),HOST,TARGET)_OUT_INTERMEDIATE_LIBRARIES), \
     , \
     , \
@@ -107,13 +113,15 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_executables), \
     $(prebuilt_is_host), \
     EXECUTABLES, \
-    $(prebuilt_module_tags))
+    $(prebuilt_module_tags), \
+    $(prebuilt_required_modules))
 
 $(call auto-prebuilt-boilerplate, \
     $(prebuilt_java_libraries), \
     $(prebuilt_is_host), \
     JAVA_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     , \
     javalib.jar)
@@ -123,6 +131,7 @@ $(call auto-prebuilt-boilerplate, \
     $(prebuilt_is_host), \
     JAVA_LIBRARIES, \
     $(prebuilt_module_tags), \
+    $(prebuilt_required_modules), \
     , \
     true, \
     javalib.jar)
@@ -134,3 +143,4 @@ prebuilt_java_libraries :=
 prebuilt_static_java_libraries :=
 prebuilt_is_host :=
 prebuilt_module_tags :=
+prebuilt_required_modules :=
-- 
1.7.4.1

